src/irmin-pack/irmin_pack.ml

```diff
@@ -793,6 +795,23 @@ module Pack (K : Irmin.Hash.S) = struct
       append t k v >|= fun () -> k
 
     let unsafe_add t k v = append t k v
+
+    let integrity_check ?(progress=(fun _ -> ())) t =
+      let count = ref 0 in
+
+      t.pack.index
+      |> Index.iter (fun k (_, _, m) ->
```

Comment:
I'm not very fond of chaining computation with `|>` for long pipeline chains. I think that in that case, giving a name to temporary created values could help reading and understanding that code.

Reply:
Long in the sense of extended computation, or many stages in the pipeline? Either way, I'm happy to change to `let ... in`.